Assets/Scripts/Actors/ActorComponents/ActorPhysics.cs

```diff
@@ -43,113 +48,135 @@ public float groundedMoveSpeed
-
-	// Model Info
-	[Space(10)][Header("Model Info")]
-	public Transform model;
+	#endregion
+
+	#region Collisions
+	[Space( 10 ), Header( "Collisions" )]
+	public Collider bumper;
+	public Collider lifter;
+	public Collider climbBumper;
+	private Transform _bumperTransform;
+	private Rigidbody _bumperRigidbody;
+	#endregion
+
+	#region Model Info
+	[Space( 10 ), Header( "Model Info" )]
 	[SerializeField] float modelTurnSpeed = 7f;
-	protected Vector3 modelOffset;
-
-	// Slope Checking
-	[Space(10)][Header("Slope Checking")]
-	[SerializeField] float groundSlopeCheckRadius = 0.2f;
-	[SerializeField] float groundSlopeCheckHeight = 0.7f;
-	float groundSlopeCapsuleWidth = 0.4f;
-	float groundSlopeCastDist = 0.3f;
+	[Range( 0.0f, 1.0f )]
+	[SerializeField] float lookDirLerp = 0.002f;
 
-	float groundSlopeFallSpeed = -30f;
+	private Vector3 _lastPos;
+	#endregion
 
-	protected float groundSlopeSpeedMod = 1f;
-	float groundSlopeSpeedModeDelta = 2.5f;
-
-	[SerializeField] MinMaxF slopeLimits = new MinMaxF( 0.28f, 0.72f );
-
-	// Jumping
-	[Space(10)][Header("Jumping")]
+	#region Jumping
+	[Space( 10 ), Header( "Jumping" )]
 	[SerializeField] public float jumpForce = 8.5f;
 
-	[SerializeField] float jumpCheckDistance = 1.3f;
-	[SerializeField] Vector3 groundPosOffset = new Vector3(0f, -1f, 0f);
-	
-	[SerializeField] float jumpCheckRadius = 1.2f;
+	[SerializeField] float jumpCheckDistance = 1.0f;
+	[SerializeField] float jumpCheckRadius = 0.5f;
+
 	[SerializeField] LayerMask jumpLayer = 0;
 	[SerializeField] float minJumpDot = 0.4f;
 
 	[SerializeField] float jumpColCheckTime = 0.5f;
-	float jumpColCheckTimer = 0.0f;
+	float _jumpColCheckTimer = 0.0f;
+
+	[SerializeField] float lateJumpTime = 0.5f;
+	bool _lateJump = false;
 
-	[SerializeField] float lateJumpTime = 0.2f;
-	float lateJumpTimer = 0.0f;
+	[Tooltip( "The minimum delay between jump checks. Use this to prevent the player from immediately colliding with the ground after jumping." )]
+	[SerializeField] float jumpCheckDelay = 0.1f;
+	bool _jumpCheckDelay = false;
 
-	// Rolling
-	[Space(10)][Header("Rolling")]
+	bool _isOnGround = false;
+	#endregion
+
+	#region Rolling
+	[Space( 10 ), Header( "Rolling" )]
 	[SerializeField] float rollTime = 1f;
 	[SerializeField] float rollCooldownTime = 1f;
 	float rollCooldownTimer = 0f;
+	#endregion
 
-	// Climbing
-	[Space(10)][Header("Climbing")]
+	#region Climbing
+	[Space( 10 ), Header( "Climbing" )]
 	[SerializeField] LayerMask climbLayer = (LayerMask)0;
-	[SerializeField] float climbCheckDistance = 0.5f;
 	[SerializeField] float climbCheckRadius = 0.7f;
-	[SerializeField] float surfaceHoldForce = 0.1f;
 
 	[SerializeField] float climbCheckTime = 0.2f;
 	float climbCheckTimer = 1f;
 
 	Transform climbSurface = null;
-
-	public bool isGrabbing
-	{
-		get{ return WadeUtils.ValidAxisInput("Grab"); }
-	}
+	ClimbableTag climbTag;
+	Vector3 climbSurfaceNormal = Vector3.zero;
+	Vector3 climbSurfaceRight = Vector3.zero;
```

Comment:
If we have climbSurface saved can't we just get these when we need them? Would be easy to write your z-in/out-of-wall check to be reusable.